src/pairings.rs

```diff
@@ -482,31 +483,33 @@ impl Group for Gt {
 /// conjunction with the [`multi_miller_loop`](crate::multi_miller_loop)
 /// function provided by this crate.
 ///
-/// Requires the `alloc` and `pairing` crate features to be enabled.
+/// Requires the `pairing` crate feature to be enabled.
 pub struct G2Prepared {
     infinity: Choice,
-    coeffs: Vec<(Fp2, Fp2, Fp2)>,
+    coeffs: [G2Coeffs; G2_PREPARED_COEFFS],
```

Comment:
After this change, `coeffs` is 19,584 bytes, which is maybe too large to just store on the stack? If someone's using this in e.g. a batch verification context then they are probably placing these directly into a `Vec<(G1Affine, G2Prepared)>`, but this feels like a change that needs to be documented somewhere. Thoughts @ebfull?

It would be a good idea to separate out the no-alloc changes into a separate PR, to faciliate merging the zeroization changes.

Reply:
I created a new PR with just the zeroize changes and no additional history. Since G2Prepared and multi_miller_loop are only supported on alloc at the moment, I suppose a Vec could be used when alloc is enabled, and an array otherwise.